spec/lib/fca/query_spec.rb

```diff
@@ -92,26 +91,129 @@
     end
 
     describe '.values' do
-      %w[4 9].each do |status|
-        context "active adviser status=#{status}" do
-          let(:line) do
-            "100039|Crawfords|23|1|#{status}|Stanton House|41 Blackfriars Road|||Salford|Lancashire|M3|7DB|44|0161|832 5366|44|0161|832 1829|Cancelled|20090521|20011201|CRAWFORDS|20090521|".force_encoding('ISO-8859-1')
+      def new_query(table)
+        FCA::Query.new(delimeter: '|', table: table)
+      end
+
+      def new_row(line, table)
+        FCA::Row.new(line, delimeter: '|', table: table)
+      end
+
+      describe 'firms' do
+        let(:table) { :lookup_firms }
+
+        let(:query_table) { new_query(table) }
+
+        def line(status)
+          "100015|Saffron Building Society|Incorporated under Building Societies Act 1986|Regulated|Not hold and not control client money|Saffron House|1a Market Place|||Saffron Walden|Essex|CB10 1HX|44|01799522211|#{status}|20011201|20010704|SAFFRONBUILDINGSOCIETY|20210511|IP00485B|||"
+        end
+
+        allowed = ['Authorised', 'EEA Authorised', 'Registered']
```

Comment:
Might make sense to enumerate the `FIRM_APPROVED_STATUSES` from the `Row` instead? Any additions or removals over there will be covered here then.